### PR TITLE
ASM-5448:Added tear down support to the Brocade puppet module

### PR DESCRIPTION
**Pull request by arunrordell**

### CODE REVIEW

File: lib/puppet/provider/brocade_zone/brocade_zone.rb [hunk 1]
```diff
@@ -41,13 +41,20 @@ def destroy
   def exists?
     Puppet.debug("Inside create zone exists block")
     initialize_resources
+    require 'pry'
+    binding.pry
```
Comment (ripienaar) [inline lib/puppet/provider/brocade_zone/brocade_zone.rb]: was this included here intentionally?

Comment (arunrordell) [inline lib/puppet/provider/brocade_zone/brocade_zone.rb]: Oh sorry. It is just by mistake.

File: manifests/createzone.pp [hunk 2]
```diff
@@ -22,46 +25,59 @@
...
 #    server_wwn => '50:00:d3:10:00:5f:a5:00',
 #    zoneset => 'Zone_Config',
+#    ensure => 'present'
 #  }
 #
 define brocade::createzone (
   $storage_alias,
   $server_wwn,
-  $zoneset,
+  $zoneset,$ensure,
```
Comment (gavin-scott) [inline manifests/createzone.pp]: Can you put `$ensure` on its own line?

File: manifests/createzone.pp [hunk 3]
```diff
@@ -22,46 +25,59 @@
...
+      configstate => "enable",
+  }
 
+  if $ensure == 'present'{
     Brocade_zone["$name"]
     -> Brocade_zone_membership["$name:$storage_alias"]
     -> Brocade_zone_membership["$name:$server_wwn"]
     -> Brocade_config_membership["$zoneset:$name"]
     -> Brocade_config["$name:$zoneset"]
+  }
```
Comment (gavin-scott) [inline manifests/createzone.pp]: Please pull the `else` up here like `} else {`

File: manifests/createzone.pp [hunk 4]
```diff
@@ -22,46 +25,59 @@
...
+  else
+  {
+    Brocade_config_membership["$zoneset:$name"]
+    -> Brocade_zone_membership["$name:$storage_alias"]
+    -> Brocade_zone_membership["$name:$server_wwn"]
+    -> Brocade_zone["$name"]
+    -> Brocade_config["$name:$zoneset"]
+  }
+
 }
```
Comment (gavin-scott) [inline manifests/createzone.pp]: Please delete the empty lines.

**Comment by arunrordell** [comment]
@gavin-scott I have updated the code

**Pull request merged**